src/ape/api/providers.py

```diff
@@ -704,6 +706,19 @@ def get_transactions_by_block(self, block_id: BlockID) -> Iterator:
         for transaction in block.get("transactions"):  # type: ignore
             yield self.network.ecosystem.create_transaction(**transaction)  # type: ignore
 
+    def stream_request(self, method, params, iter_path="result.item"):
+        payload = {"jsonrpc": "2.0", "id": 1, "method": method, "params": params}
+        results = ijson.sendable_list()
+        coro = ijson.items_coro(results, iter_path)
+
+        resp = requests.post(self.connection_str, json=payload, stream=True)
```

Comment:
There is a couple issues with this I think.

1.) This assumes that the `Web3Provider` is HTTP based.
2.) `connection_str` is only part of `UpstreamProvider`.

We can get rid of UpstreamProvider and make `connection_str` part of the root ProviderAPI class. This would be a breaking change but I am ok with it

To address the first point, we can verify that connection_str starts with http:// or https:// first or else raise an `APINotImplementedError` saying that subclass needs to implement its own stream_request

Reply:
i see what you mean, i initially put it into `GethProvider` which explicitly [uses HTTPProvider](https://github.com/ApeWorX/ape/blob/74ec1f42481f0a8d940e3c085ac13fbf0de57b4a/src/ape_geth/providers.py#L176). maybe i should move it back?